src/manage_sql_nvts.c

```diff
@@ -1496,6 +1497,65 @@ manage_update_nvt_cache_osp (const gchar *update_socket)
                " VALUES ('checked_preferences', 1)"
                " ON CONFLICT (name) DO UPDATE SET value = EXCLUDED.value;");
         }
+
+      /* Update scanner preferences */
+      connection = osp_connection_new (update_socket, 0, NULL, NULL, NULL);
+      if (!connection)
+        {
+          g_warning ("%s: failed to connect to %s (3)",
+                    __FUNCTION__, update_socket);
+          return -1;
+        }
+
+      scanner_prefs = NULL;
+      if (osp_get_scanner_details (connection, NULL, &scanner_prefs))
+        {
+          g_warning ("%s: failed to get scanner preferences", __FUNCTION__);
+          osp_connection_close (connection);
+          return -1;
+        }
+      else
+        {
+          GString *prefs_sql;
+          GSList *point;
+          int first;
+
+          point = scanner_prefs;
+          first = 1;
+
+          osp_connection_close (connection);
+          prefs_sql = g_string_new ("INSERT INTO nvt_preferences (name, value)"
```

Comment:
I think this needs to handle the case where the scanner preference exists already.

Reply:
This should be handled by the `ON CONFLICT (name) ...` clause and the I've run the enclosing if block multiple times without getting any duplicates.

Reply:
Oh I missed that, nice.